include/gmsh.tcc

```diff
@@ -138,13 +143,25 @@ void GMSH::compute_stresses() {
 
   //! Loop through the points to get vertical and horizontal stresses
   //! Note that tau (shear stress) is assumed 0
-  for (const auto& point : vertices_) {
-    ver_stress =
-        conv_factor * (-(max_height - point->coordinates().at(2))) * density;
+  for (const auto& point : coords_) {
```

Comment:
Only change required is this, when you find the stress, assign stress to the right material point.